src/server/blocking_controller_test.cc

```diff
@@ -21,9 +22,14 @@ using namespace std;
 using namespace std::chrono;
 using namespace testing;
 
+void FakeDependency() {
+  ShardDocIndices sdi;  // fake dependency to enfore linking
+}
+
```

Comment:
@royjacobson There surely must be a better way to explicitly set dependencies.

The issue is kinda dull: the transaction library is now incomplete. But I can't decouple it from the server library and nor want I to move ALL search files to the transaction library. The only place where we use them separately is this test. I don't see any more sense in this separation

Reply:
enforce (typo)
But I'm not sure what problem this take dependency enforcement solves, as we specify linking in the CMake files. Is the linker stripping away symbols it shouldn't?

Reply:
Seems like. It worked without before even though there was a call to the ShardIndides constructor. Now it stopped because I added a GetMemory call (stopped form both)

I'm not an expert in compilation and this looks very much like magic to me... How did it work before? Why does it now 🤷🏻 I hope Roy can help 😌

Reply:
I think it's a dependency cycle between `dragonfly_lib` and `dfly_transaction`. `dragonfly_lib` should be dependent on `dfly_transaction` but it seems it's using two functions (`ShardDocIndices::ShardDocIndices`, `ShardDocIndices::GetUsedMemory`) that are inside `dragonfly_lib` and it's causing weird linking errors down the line.

Reply:
There is, I mentioned it before. This split is very artificial and somewhat annoying in this context

> But I can't decouple it from the server library and nor want I to move ALL search files to the transaction library

Reply:
As long as both targets are always used together, it should not cause any issues

Reply:
Ok, what about moving `engine_shard_set.cc` into `dragonfly_lib`?